### PR TITLE
#104 navbar user icon & some bug fixes

### LINKED ISSUE #53: Flash of unstyled content on My Trecipe page
There is flash of unstyled content on the Trecipe logo. To be addressed in the future 

<ins>**Steps to reproduce**:</ins>

1. Go to My Trecipe page
2. Give all your attention to the Trecipe logo/font on the top left
3. Refresh the page  

_Result_: The logo has a different font-family for a brief moment.
_Expected_: The font family of the logo should remain the same

The gif is slowed down to make it more obvious:

fous

### LINKED ISSUE #104: Remove login link from navbar and show user icon with dropdown menu

### LINKED ISSUE #105: Reset scrolling to top when going from page to page
Repro step:
Go from Trecipe Page into Map page, you will see the scrolling is not set to top.

### PR DESCRIPTION
**Pull request by knox153**
This PR addresses a few things 🛠

## User icon on navigation barrr
Closes #104 

**When user is signed in:**

---
image

--- 

**When user is signed out:** 

---
image

---

## Fix rating overflow on destination card 
Rating no longer overflow when the address is long:
image

## Scroll to top on page navigation
Closes #105 
Added a new component `ScrollToTop` to scroll the page to top on page navigation, done according to the [official doc](https://reactrouter.com/web/guides/scroll-restoration)

## Load Google font on head element 
Closes #53  
Importing google font in stylesheet can be slow and might cause flash of unstyled text depending on browser. Now we load Google font in head element (in index.html) to fix flash of unstyled text.

### CODE REVIEW

File: packages/client/src/components/CardMenu/CardMenu.tsx [hunk 1]
```diff
@@ -1,13 +1,17 @@
 import React from 'react';
 import { Button } from '../Button/Button';
 import { Menu, MenuItem } from '../Menu/Menu';
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
 
 /**
  * CardMenuProps
  * menuItem: List of menu items (each with onClick callback) to display in menu
```
Comment (shizuko-akamoto) [inline packages/client/src/components/CardMenu/CardMenu.tsx:8]: Let's also add in the new props to docs

File: packages/client/src/components/LogoutButton/LogoutButton.tsx [hunk 2]
```diff
@@ -6,6 +6,8 @@ import { RootState } from '../../redux';
 import { logout } from '../../redux/User/action';
 import { withRouter, RouteComponentProps } from 'react-router-dom';
 
+// TODO: remove this file if this is not needed for responsive design
```
Comment (shizuko-akamoto) [inline packages/client/src/components/LogoutButton/LogoutButton.tsx:9]: What do you mean by this hahaha. For mobile we show a hamburger menu with all the links right

File: packages/client/src/components/Rating/ratingBar.scss [hunk 3]
```diff
@@ -1,6 +1,10 @@
 @import '../../components/variables';
 @import '../../components/mixins';
 
+.rating-bar {
```
Comment (shizuko-akamoto) [inline packages/client/src/components/Rating/ratingBar.scss:4]: Beautiful

File: packages/client/src/components/LogoutButton/LogoutButton.tsx (hunk 2 above)
Comment (knox153) [reply packages/client/src/components/LogoutButton/LogoutButton.tsx:9]: lol some website has dedicated logout button on mobile view 😛, but we are not doing that, let me remove it

**Pull request merged**